### PR TITLE
233 added email and Sms services

### PR DESCRIPTION
**Pull request by aXayitov**

### CODE REVIEW

**Review by Boburm1rzo — APPROVED** [review]

File: CheckDrive.Api/CheckDrive.Api/Extensions/DependencyInjection.cs [hunk 1]
```diff
@@ -192,5 +193,29 @@ private static void AddAuthorization(IServiceCollection services)
...
         }
 
+        private static void AddFluentEmail(IServiceCollection services, IConfiguration configuration)
+        {
+            var emailSettings = configuration.GetSection(EmailConfiguration.SectionName).Get<EmailConfiguration>();
+
+            if (emailSettings is null)
+            {
+                throw new InvalidOperationException("Configuration values for email did not load correctly.");
+            }
```
Comment (FirdavsAX) [inline CheckDrive.Api/CheckDrive.Api/Extensions/DependencyInjection.cs:203]: you can Use there null coalescing operators : ?? throw new InvalidOpertaionException to code reduction, but its not mandatory

File: CheckDrive.Api/CheckDrive.Infrastructure/Email/EmailService.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using FluentEmail.Core;
+
+namespace CheckDrive.Infrastructure.Email;
+public class EmailService : IEmailService
+{
+    private readonly Dictionary<EmailType, string> templateNames = new()
+    {
+       { EmailType.EmailConfirmation, "EmailConfirmationTemplate.cshtml" },
+       { EmailType.ForgotPassword, "ForgotPasswordTemplate.cshtml" },
+       { EmailType.DebtNotification, "DebtNotificationTemplate.cshtml" }
```
Comment (FirdavsAX) [inline CheckDrive.Api/CheckDrive.Infrastructure/Email/EmailService.cs:11]: You have not a  DebtNotification template in this project . You can remove this Element

File: CheckDrive.Api/CheckDrive.Infrastructure/Email/Factories/EmailMetadataFactory.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace CheckDrive.Infrastructure.Email.Factories;
+
+public class EmailMetadataFactory : IEmailMetadataFactory
+{
+    private readonly Dictionary<EmailType, string> emailSubjects = new()
+    {
+        { EmailType.EmailConfirmation, "Email Confirmation" },
+        { EmailType.ForgotPassword, "Password Reset" },
+        { EmailType.DebtNotification, "Debt warning" }
```
Comment (FirdavsAX) [inline CheckDrive.Api/CheckDrive.Infrastructure/Email/Factories/EmailMetadataFactory.cs:13]: You have not DebtNotification template view. You can remove that, its not necessary

File: CheckDrive.Api/CheckDrive.Infrastructure/Email/Factories/IEmailMetadataFactory.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿using CheckDrive.Infrastructure.Models;
+
+namespace CheckDrive.Infrastructure.Email.Factories;
+
+public interface IEmailMetadataFactory
+{
+    EmailMetadata Create(EmailType emailType, string to,
+    UserInfo? userInfo = null, string? callbackUri = null,
+    List<string>? cc = null);
```
Comment (FirdavsAX) [inline CheckDrive.Api/CheckDrive.Infrastructure/Email/Factories/IEmailMetadataFactory.cs:9]: There code is not readeble, please use put each parameter to separate row.

File: CheckDrive.Api/CheckDrive.Infrastructure/Models/EmailType.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CheckDrive.Infrastructure.Models
+{
+    public enum EmailType
+    {
+        ForgotPassword,
+        EmailConfirmation,
+        DebtNotification
```
Comment (FirdavsAX) [inline CheckDrive.Api/CheckDrive.Infrastructure/Models/EmailType.cs:13]: you have not DebtNotification template, please remove it in enum

**Review by FirdavsAX — CHANGES_REQUESTED** [review]

**Review by FirdavsAX — APPROVED** [review]

**Review by Mirazyzz — APPROVED** [review]
Looks good!

**Pull request merged**